docs/dbt-performance.md

```diff
@@ -0,0 +1,114 @@
+
+![dbt Model Run Summary](images/dbt_run_summary.png)
+
+For a quick visual reference of which models take up the most time in a run, click on the "Model Timiing" tab. If you hover over a model you will be shown the specific timing.
+
+![dbt Model Timing Graph](images/dbt_model_timing.png)
+
+## Solutions: How to Tackle dbt Performance Issues
+Now that you've identified which models might need optimization, it's time to figure out how to get them to run faster. These options are roughly in order of bang-for-buck in most situations.
+
+### 1. Adjust Frequency and Break Up Runs
+It's common for end-users to say they want the freshest data (who doesn't?) but in practice require a much lower frequency of refreshing. To gain an understand of the real-world needs it's helpful to see the frequency with which end-users actually view reporting and to consider the time scales involved. If someone only cares about monthly results, for example, you can *in theory* have a 30 day frequency for model runs.
+It's also quite common to have parts of the data be relatively static, and only need to be refreshed occasionally whereas other parts of the data might change much more often.
+An easy way to break up model runs is by using dbt tags.
+```yaml
+models:
+    foo_model:
+        +tags:
+            - "daily"
+            - "staging"
+    bar_model:
+        +tags: "hourly"
+    baz_)_model:
+        +tags: "weekly"
+```
+
+In this case you can run certain models with:
+`dbt run --select tag:daily`
+Of course this works in dbt Cloud as well!
+
+For more information refer to the [dbt tags documentation](https://docs.getdbt.com/reference/resource-configs/tags).
+
+
+### 2. Materialization Matters
+
+For a good comparison of materialization options and their trade-offs see the [Materialization Best Practices](https://docs.getdbt.com/guides/best-practices/materializations/2-available-materializations) section of the dbt docs.
+
+**Views**: Are a trade-off between build performance and read/reporting performance. In cases where you are using a BI tool, you should almost always use table materializations unless data storage size is an issue or refresh frequency is so high that cost or scheduling conflicts become a problem. In cases where performance at time of reporting is not an issue (say, you are generating an aggregated report on a monthly basis) then views can be a great way to cut run time. Another case where views can be a good option is with staging data of relatively small size, where your queries are relatively light-weight and you want to ensure fresh data without having to configure separate runs for those models.
+
+### 3. Incremental Models
```

Comment:
This feels like it should be at the same level of organization as **Views** (i.e., under the "Materialization Matters" subhead)

Reply:
@AeriShan-ODI do you agree with this, or want to go in as is?